src/WebJobs.Script/Rpc/Configuration/WorkerConfigFactory.cs

```diff
@@ -220,5 +239,83 @@ internal string GetExecutablePathForJava(string defaultExecutablePath)
                 return Path.GetFullPath(Path.Combine(javaHome, "bin", defaultExecutablePath));
             }
         }
+
+        internal bool IsHydrationNeeded(string workerPath)
+        {
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return false;
+            }
+
+            return workerPath.Contains(LanguageWorkerConstants.OSPlaceholder) ||
+                    workerPath.Contains(LanguageWorkerConstants.ArchitecturePlaceholder) ||
+                    workerPath.Contains(LanguageWorkerConstants.RuntimeVersionPlaceholder);
+        }
+
+        internal string GetHydratedWorkerPath(WorkerDescription description)
+        {
+            string workerPath = description.GetWorkerPath();
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return null;
+            }
+
+            OSPlatform os = _systemRuntimeInformation.GetOSPlatform();
+
+            Architecture architecture = _systemRuntimeInformation.GetOSArchitecture();
+            string version = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeVersionSettingName);
+            if (string.IsNullOrEmpty(version))
+            {
+                version = description.DefaultRuntimeVersion;
+            }
+
+            ValidateWorkerPath(workerPath, description.Language, os, architecture, version);
+
+            return workerPath.Replace(LanguageWorkerConstants.OSPlaceholder, os.ToString())
+                             .Replace(LanguageWorkerConstants.ArchitecturePlaceholder, architecture.ToString())
+                             .Replace(LanguageWorkerConstants.RuntimeVersionPlaceholder, version);
+        }
+
+        internal void ValidateWorkerPath(string workerPath, string language, OSPlatform os, Architecture architecture, string version)
```

Comment:
move validation helpers to: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.Grpc/Abstractions/WorkerDescriptionExtensions.cs